### PR TITLE
Promote RemainingItemCount to beta in Feature Gates docs

### PR DESCRIPTION
**Pull request by haardikdharma10**
Fixes: #27944

### CODE REVIEW

**Comment by haardikdharma10** [comment]
> Should we also update https://kubernetes.io/docs/reference/using-api/api-concepts/#retrieving-large-results-sets-in-chunks to mention that responses include a remaining item count by default?

Sure @sftim, where would it fit exactly?

**Comment by irvifa** [comment]
I think it's still within referenced issue, can we add this within this PR as well?

**Comment by kbhawkey** [comment]
Hi @haardikdharma10 .
I think I would put a comment in this page and section:
https://kubernetes.io/docs/reference/using-api/api-concepts/#retrieving-large-results-sets-in-chunks

From the original PR (https://github.com/kubernetes/kubernetes/pull/81682):
```
remainingItemCount is the number of subsequent items in the list which are not included in this list response ... The intended use of the remainingItemCount is *estimating* the size of a collection. 
```

**Comment by haardikdharma10** [comment]
@sftim, ping for a review:)

File: content/en/docs/reference/using-api/api-concepts.md [hunk 1]
```diff
@@ -192,7 +192,7 @@ For example, if there are 1,253 pods on the cluster and the client wants to rece
...
+Note that the `resourceVersion` of the list remains constant across each request, indicating the server is showing us a consistent snapshot of the pods. Pods that are created, updated, or deleted after version `10245` would not be shown unless the user makes a list request without the `continue` token.  This allows clients to break large requests into smaller chunks and then perform a watch operation on the full set without missing any updates. `remainingItemCount` is the number of subsequent items in the list which are not included in this list response. If the list request contained label or field selectors, then the number of remaining items is unknown and the field will be left unset and omitted during serialization. If the list is complete (either because it is not chunking or because this is the last chunk), then there are no more remaining items and this field will be left unset and omitted during serialization. Servers older than v1.15 do not set this field. The intended use of the remainingItemCount is estimating the size of a collection.
```
Comment (tengqm) [inline content/en/docs/reference/using-api/api-concepts.md:195]: Please wrap this long line manually. Such long lines are difficult for change tracking.

Comment (tengqm) [reply content/en/docs/reference/using-api/api-concepts.md:195]: > The intended use of the remainingItemCount is estimating the size of a collection. 

The intended use of the `remainingItemCount` is estimating the size of a collection.

Comment (sftim) [reply content/en/docs/reference/using-api/api-concepts.md:195]: > Servers older than v1.15 do not set this field.

We don't need to document that, as a control plane older than v1.15 is out of support.

> and the field will be left unset and omitted during serialization.

I'd write “and the API server does not include a `remainingItemCount` in its response”.

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-28066--kubernetes-io-master-staging.netlify.app/docs/reference/using-api/api-concepts/#retrieving-large-results-sets-in-chunks

File: content/en/docs/reference/using-api/api-concepts.md [hunk 2]
```diff
@@ -192,7 +192,7 @@ For example, if there are 1,253 pods on the cluster and the client wants to rece
...
+Note that the `resourceVersion` of the list remains constant across each request, indicating the server is showing us a consistent snapshot of the pods. Pods that are created, updated, or deleted after version `10245` would not be shown unless the user makes a list request without the `continue` token.  This allows clients to break large requests into smaller chunks and then perform a watch operation on the full set without missing any updates. `remainingItemCount` is the number of subsequent items in the list which are not included in this list response. If the list request contained label or field selectors, then the number of remaining items is unknown and the API server does not include a `remainingItemCount` field in its response. If the list is complete (either because it is not chunking or because this is the last chunk), then there are no more remaining items and the API server does not include a `remainingItemCount` field in its response. The intended use of the `remainingItemCount` is estimating the size of a collection.
```
Comment (kbhawkey) [inline content/en/docs/reference/using-api/api-concepts.md:195]: Hi @haardikdharma10 .
I think the description is okay. Thanks.
What do you think about wrapping the text?
Does this content belongs in a separate paragraph?

Comment (haardikdharma10) [reply content/en/docs/reference/using-api/api-concepts.md]: Hey @kbhawkey, i have wrapped the text manually. The content also seemed to belong in a separate paragraph, fixed that too.

**Comment by tengqm** [comment]
based on previous feedbacks, I'm gonna kick this in.
/lgtm
/approve

**Comment by sftim** [comment]
Thanks @haardikdharma10 & @tengqm

**Pull request merged**